### PR TITLE
docs: list dependencies' license

### LINKED ISSUE #22874: Publishing the copyrights and licenses of the third party libraries
Some of the third party libraries that Envoy uses have license terms that demand that the copyright or the license itself is included in the software and/or documentation.

For example, [the nghttp2 library uses the MIT license](https://github.com/nghttp2/nghttp2/blob/master/COPYING). It contains this sentence: "Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to [...], subject to the following conditions: The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software."

How does Envoy comply with this legal requirement?

I think the copyright notice and the license of nghttp2 (and other third-party libraries) should appear somewhere in Envoy's documentation. There is already a [table that lists all third-party libraries](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/external_deps). A similar table that contains a list of the external dependencies, their licenses and (if necessary) also the complete license texts would solve this problem.

### PR DESCRIPTION
**Pull request by zhxie**
Signed-off-by: Xie Zhihao <zhihao.xie@intel.com>

Commit Message: docs: list dependencies' license
Additional Description:

TODO:

- BoringSSL (FIPS): unknown repo
- liburing: dual license needs to be chosen by code owners
- skywalking-data-collect-protocol: license in README only
- zstd: dual license needs to be chosen by code owners
- wavm: version not found
- V8: unknown repo
- Chromium's trace event headers: license not found
- Common Expression Language (CEL) C++ library: version not found
- Kafka (server binary): unknown repo
- Intel Dlb: unknown repo

Risk Level: Low
Testing: N/A
Docs Changes: Yes
Release Notes: N/A
Platform Specific Features: N/A
Partial fix for #22874

### CODE REVIEW

**Comment by zhxie** [comment]
@phlax can you check in advance if it satisfies.

Link to external deps: https://storage.googleapis.com/envoy-pr/d3a7e84/docs/intro/arch_overview/security/external_deps.html

File: tools/docs/generate_external_deps_rst.py [hunk 1]
```diff
@@ -126,7 +126,14 @@ def main():
         name = rst_link(project_name, project_url)
         version = rst_link(render_version(v['version']), get_version_url(v))
         release_date = v['release_date']
-        dep = Dep(name, project_name.lower(), version, cpe, release_date)
+        license = v.get('license', '')
+        if license == 'N/A':
```
Comment (phlax) [inline tools/docs/generate_external_deps_rst.py:130]: im wondering if we need to accomodate N/A if the license is not compulsory (perhaps it should be)

**Review by phlax — COMMENTED** [review]
@zhxie looks great, thanks for picking this up so quickly

i dont see `license_path` used so im guessing we didnt need it - which i think is also simpler

**Comment by phlax** [comment]
cc @moderation

File: tools/docs/generate_external_deps_rst.py (hunk 1 above)
Comment (zhxie) [reply tools/docs/generate_external_deps_rst.py:130]: I am not sure if license is compulsory or not, but in my opinion, I think license should have the same priority as CVE. In fact, CVE is also not compulsory at the moment. It is a simple copy-paste, and I can change it if needed.

Comment (phlax) [reply tools/docs/generate_external_deps_rst.py:130]: yep - lets go with not making it compulsory, at least for now - but lets drop the N/A - rather if its not filled in just dont add it

Comment (zhxie) [reply tools/docs/generate_external_deps_rst.py]: Fixed.

**Comment by moderation** [comment]
Wondering what is the driver for this. Is there there is an end-user asking for this or is this a nice to have?

Before - https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/external_deps
After - https://storage.googleapis.com/envoy-pr/4c0cb9d/docs/intro/arch_overview/security/external_deps.html

**Comment by phlax** [comment]
> Wondering what is the driver for this. Is there there is an end-user asking for this or is this a nice to have?

see https://github.com/envoyproxy/envoy/issues/22874 for related discussion - but the issue has come up recently on slack also

mostly license compliance - ie - do whatever you want as long as you "include" this license

**Comment by phlax** [comment]
> Before/After ...

some rows are a bit wide on non-breaking cols  - perhaps we could do some truncation or similar - altho probs outside scope of this PR

https://storage.googleapis.com/envoy-pr/a07d03d/docs/intro/arch_overview/security/external_deps.html?highlight=external#envoy-wasm-runtime-wavm

**Review by phlax — APPROVED** [review]
lgtm, thanks @zhxie 

lets land this and follow up with the missing licenses etc

**Pull request merged**